### PR TITLE
Implemented #1721: see users who forked/starred/watched a repository

### PR DESCRIPTION
**Pull request by soudy**
The logic and templates are there and are accessible through their paths (/repo/stars, /repo/forks, /repo/watchers). It is however not implemented in the repo's UI.

### CODE REVIEW

File: cmd/web.go [hunk 1]
```diff
@@ -514,6 +514,9 @@ func runWeb(ctx *cli.Context) {
 		m.Get("/labels/", repo.RetrieveLabels, repo.Labels)
 		m.Get("/milestones", repo.Milestones)
 		m.Get("/branches", repo.Branches)
+		m.Get("/stars/?:index", middleware.RepoRef(), repo.Stars)
+		m.Get("/watchers/?:index", middleware.RepoRef(), repo.Watchers)
+		m.Get("/forks", middleware.RepoRef(), repo.Forks)
```
Comment (tboerger) [inline cmd/web.go]: Why did you not add an index here as well?

File: public/ng/less/gogs/repository.less [hunk 2]
```diff
@@ -795,3 +795,84 @@
         padding-top: 15px;
     }
 }
+
+#stars, #watchers, #forks {
```
Comment (tboerger) [inline public/ng/less/gogs/repository.less]: Does it really make sense to add that much custom CSS for this simple kind of pages?

File: cmd/web.go (hunk 1 above)
Comment (soudy) [inline cmd/web.go]: Forks show less data than stars and watchers, so it all fits pretty well on one page.

**Comment by tboerger** [comment]
The templates got in general pretty bad indention.

**Comment by soudy** [comment]
@tboerger could you elaborate on what's wrong with the indention?

File: cmd/web.go (hunk 1 above)
Comment (tboerger) [inline cmd/web.go]: I think that toally depends on the size of the gogs install. I would suggest to add paging but display it only for a bigger number of rows.

**Comment by tboerger** [comment]
> @tboerger could you elaborate on what's wrong with the indention?

Just look at the diff? Maybe https://github.com/gogits/gogs/pull/1725/files#diff-6e6e73dad0326f79bf8064a4851cad92R35

**Comment by Fastidious** [comment]
Looks sloppy to me, I agree with @tboerger, and a lot of new CSS.

**Comment by soudy** [comment]
@tboerger strangely enough it looked fine in my text editor, thanks
@Fastidious these are 3 new pages, some new CSS was needed

**Comment by tboerger** [comment]
@soudy thankfully to the CSS framework I'm pretty sure we don't need that much heavily customized CSS.

**Comment by tboerger** [comment]
I still dislike the indention of that, but at the end it's the decision of @Unknwon.

I'm absolutely sure that this CSS is not required, especially the pagination CSS must be already somewhere present because it's not the first page including pagination.

You are still referring to some old templates?

Beside of that you should create proper commit messages. Maybe sonsider some git rebase instead of git merge commands, that way you can avoid commit messages like `	Merge branch 'develop' of https://github.com/soudy/gogs into develop`. At the end of a pull request it makes sense to squash the commits.

**Comment by soudy** [comment]
@tboerger thanks for the advice, I'll see what I can do with the CSS

**Comment by Unknwon** [comment]
@soudy very much thank to your PR!

However, you should really discuss on [Gitter.im](https://gitter.im/gogits/gogs) as big red notice here: https://github.com/gogits/gogs#notices

Because some details are not shown in the surface. For example, the page style you have in the PR is based on old CSS theme which is going to removed at some point.

Always ask and confirm and save very much of your time. :stuck_out_tongue_closed_eyes:

**Pull request merged**